src/pybind/mgr/cephadm/services/monitoring.py

```diff
@@ -409,6 +420,10 @@ def calculate_deps(self) -> List[str]:
         # add an explicit dependency on the active manager. This will force to
         # re-deploy prometheus if the mgr has changed (due to a fail-over i.e).
         deps.append(self.mgr.get_active_mgr().name())
+        # add dependency on ceph-exporter daemons
+        exporter_daemons = self.mgr.cache.get_daemons_by_service('ceph-exporter')
+        if exporter_daemons:
```

Comment:
nit: `exporter_daemons` is a list so iterating over an empty list shouldn't be a problem